app/controllers/mixins/actions/host_actions/misc.rb

```diff
@@ -142,6 +142,17 @@ def process_hosts_generic(hosts, task, display_name)
           end
         end
 
+        def process_hosts_generic(hosts, task, display_name)
+          each_host(hosts, display_name) do |host|
+            if host.supports?(task.to_sym)
+              host.send(task.to_sym)
+              add_flash(_("\"%{record}\": %{task} successfully initiated") % {:record => host.name, :task => display_name})
+            else
+              add_flash(host.unsupported_reason(task), :error)
+            end
+          end
+        end
+
```

Comment:
NOTE I no longer need this because of the supports/is_available work @kbrock did